### PR TITLE
New package: crun-0.14.1

### PR DESCRIPTION
**Pull request by natrys**

### CODE REVIEW

**Comment by Anachron** [comment]
I'd be interested in this package as well, we probably need to figure out why those tests are failing first.

**Comment by Noah-Huppert** [comment]
All the failing test have the error: `clone: Operation not permitted`.

I wonder if there is some namespace call that crun can't access b/c it is in chroot?

Full test error log here: https://gist.github.com/Noah-Huppert/7c080aa8c2372bddeb9a5fa9769068c8

**Comment by natrys** [comment]
Sorry, I was down with fever. Yeah tests in crossbuild being wonky aside, I tried the official(?) package in my Fedora box and not all tests passed there too anyway. Perhaps it's good enough that I can verify the cross build works (e.g. in Podman) in i686/aarch64 and in musl, so I am marking it done.

File: srcpkgs/crun/template [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+hostmakedepends="pkg-config libtool python3 go-md2man"
+makedepends="libcap-devel libseccomp-devel yajl-devel"
+short_desc="Lightweight OCI runtime written in C"
+maintainer="Imran Khan <imrankhan@teknik.io>"
+license="GPL-2.0-or-later, LGPL-2.1-or-later"
+homepage="https://github.com/containers/crun"
+distfiles="https://github.com/containers/crun/releases/download/${version}/crun-${version}.tar.gz"
+checksum=a7a9f458fa4c13fc63f8e74c6ce660f9c439022cd50b4a00902c258ef08e75ff
+
+case "$XBPS_TARGET_MACHINE" in
```
Comment (xtraeme) [inline srcpkgs/crun/template]: Change this to

```
if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
 ...
fi
```

File: srcpkgs/crun/template [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	*-musl) makedepends+=" argp-standalone";;
+esac
+
+pre_configure() {
+	if [ "$CROSS_BUILD" ]; then
+		configure_args+=" --host=${XBPS_CROSS_TRIPLET}"
```
Comment (xtraeme) [inline srcpkgs/crun/template]: no need to use `pre_configure` to append to `configure_args`

Comment (natrys) [reply srcpkgs/crun/template]: Changed, thanks.

File: srcpkgs/crun/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+homepage="https://github.com/containers/crun"
+distfiles="https://github.com/containers/crun/releases/download/${version}/crun-${version}.tar.gz"
+checksum=a7a9f458fa4c13fc63f8e74c6ce660f9c439022cd50b4a00902c258ef08e75ff
+
+if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
+	makedepends+=" argp-standalone"
+fi
+
+if [ "$CROSS_BUILD" ]; then
+	configure_args+=" --host=${XBPS_CROSS_TRIPLET}"
```
Comment (sgn) [inline srcpkgs/crun/template:21]: Isn't this should be set by `build_style=gnu-configure`?

Comment (natrys) [reply srcpkgs/crun/template]: I see. Always wondered why there is both `configure` and `gnu-configure`. Fixed and upgraded to recent upstream version, if there is interest.

**Pull request closed without merging**